src/pybind/mgr/dashboard_v2/models/pool.py

```diff
@@ -0,0 +1,149 @@
+    cache_mode = CharField(choices=[(c, c) for c in 'none|writeback|forward|readonly|readforward|'
+                                                    'readproxy'.split('|')], blank=True)
+    # TODO: Replace with CephPool reference
+    tier_of = IntegerField(null=True, default=None, blank=True)
+    # TODO: Replace with CephPool reference
+    write_tier = IntegerField(null=True, default=None, blank=True)
+    # TODO: Replace with CephPool reference
+    read_tier = IntegerField(null=True, default=None, blank=True)
+    target_max_bytes = IntegerField(blank=True)
+    hit_set_period = IntegerField(blank=True)
+    hit_set_count = IntegerField(blank=True)
+    hit_set_params = JsonField(base_type=dict, editable=False, blank=True)
+    tiers = JsonField(base_type=list, editable=False, blank=True)
+    flags = JsonField(base_type=list, blank=True, default=[])
+
+    _compr_mode_choices = 'force aggressive passive none'.split(' ')
+    _compr_algorithm_choices = 'none snappy zlib zstd lz4'.split(' ')
+    compression_mode = CharField(default='none', choices=[(x, x) for x in _compr_mode_choices])
+
+    compression_algorithm = CharField(default='none',
+                                      choices=[(x, x) for x in _compr_algorithm_choices])
+    # TODO: Add validation for value to be between 0 and 1
+    compression_required_ratio = FloatField(default=0.0)
+    compression_max_blob_size = IntegerField(default=0)
+    compression_min_blob_size = IntegerField(default=0)
+
+    pool_snaps = JsonField(base_type=list, editable=False, blank=True)
+
+    application_metadata = JsonField(base_type=dict, default={}, blank=True)
+
+    @staticmethod
+    def get_all_objects(api_controller, query):
+        """":type api_controller: RESTController"""
+        result = []
+
+        osd_map = api_controller.mgr.get('osd_map')
+
+        pool_data = None
+
+        for pool_data in osd_map['pools']:
+
+            if sys.version_info < (3, 0):
+                for key, value in pool_data.items():
+                    if isinstance(value, long):
+                        pool_data[key] = int(value)
+
+            def options(key, default=None):
+                opts = pool_data.get('options', {})
+                return opts.get(key, default)
+
+            object_data = {
+                'id': pool_data['pool'],
+                'name': pool_data['pool_name'],
+                'type': {1: 'replicated', 3: 'erasure'}[pool_data['type']],  # type is an
+                                                                             # undocumented dump of
+                # https://github.com/ceph/ceph/blob/289c10c9c79c46f7a29b5d2135e3e4302ac378b0/src/osd/osd_types.h#L1035
+                'erasure_code_profile_id':
+                    (pool_data['erasure_code_profile'] if 'erasure_code_profile' in pool_data
+                     else None),
+                'last_change': pool_data['last_change'],
+                'full': 'full' in pool_data['flags_names'],
```

Comment:
We export `flags_names` already in line 112. I'd remove `full` here.

Reply:
Jap, agree.